Automattic/vip-go-mu-plugins
# Prevent saving unnecessary Jetpack post_by_email_address* DB options

## Description
This addresses the main symptom (autoloaded `post_by_email_address*` options being created per-user) of JP issue https://github.com/Automattic/jetpack/issues/35636

## Changelog Description
`cleaner_jp_pbe_options` uses `pre_update_option_post_by_email_address*` (where * is the current user ID) to prevent creation of unused NULL `post_by_email_address*` options (by returning the old_value in place of NULL, which in this case is no option at all).

## Pre-review checklist

Please make sure the items below have been covered before requesting a review:

- [ 👍 ] This change works and has been tested locally (or has an appropriate fallback).
- [ 👍 ] This change works and has been tested on a Go sandbox 
- [ 👍 ] Also tested live in a VIP test env (with this function in a standalone mu-plugin)
- [ n/a ] This change has relevant unit tests (if applicable).
- [ n/a ] This change uses a rollout method to ease with deployment (if applicable - especially for large scale actions that require writes).
- [ n/a ] This change has relevant documentation additions / updates (if applicable).
- [ 👍 ] I've created a changelog description that aligns with the provided examples.

## Pre-deploy checklist

- [ n/a ] VIP staff: Ensure any alerts added/updated conform to internal standards (see internal documentation). 

## Steps to Test

1. Check out PR.
2. Login and visit the JP setting page in wp-admin, which without this filter would spawn a post_by_email_address option
3. Do `wp option list --search="post_by_email*" --fields=option_name,option_value,autoload`
4. Verify that no new `post_by_email_address` option has been created for your logged-in user ID
5. To verify that non-NULL post_by_email_address* options aren't filtered, add this quick snippet to a sandbox and redo steps 2 & 3:
```
function create_post_by_email_option()
{
    $current_user_id = get_current_user_id();
    $option_name = 'post_by_email_address' . $current_user_id;
    update_option($option_name, 'NOTNULL');
}

add_action('admin_init', 'create_post_by_email_option');
```
